assets/js/stripe-payment-request.js

```diff
@@ -325,7 +336,6 @@ jQuery( function( $ ) {
 				requestPayerEmail: true,
 				requestPayerPhone: wc_stripe_payment_request_params.checkout.needs_payer_phone,
 				requestShipping: wc_stripe_payment_request_params.product.requestShipping,
-				displayItems: wc_stripe_payment_request_params.product.displayItems
```

Comment:
`displayItems` comes now from the backend every time the user clicks the PR button

Reply:
I've noticed that but unfortunately we can't do that now. For the product page the PR button is still rendered when the page loads and `total` [is required](https://stripe.com/docs/js/payment_request/create#stripe_payment_request-options-total) for `stripe.paymentRequest()` otherwise it will throw an error.

We need to update how the PR button is initialized on the product page and make it similar to the Cart page where the button is rendered only after getting the cart details. I was planning to do it on this PR but it required some additional changes out of the scope of this PR.

Reply:
> We need to update how the PR button is initialized on the product page and make it similar to the Cart page where the button is rendered only after getting the cart details

Would it be possible to do it on the product page? I guess we need to render the button first so it's clickable, but the total doesn't need to come from the back-end. Perhaps we can use any static total value for that. We [kinda do this](https://github.com/woocommerce/woocommerce-gateway-stripe/blob/1fc58853fa3c5c64cf1635faf443ceeaa368ff0e/client/blocks/payment-request/index.js#L42-L65) in the blocks implementation. We do a "test" payment request initialization to confirm that the method will be available. Then we update the payment request with the right totals and displayItems.

Reply:
> Would it be possible to do it on the product page? 

I think it's possible. We could initialize the PR button with a minimum amount so it's clickable right away. But still, the PR button will have two ways to initialize it. One where the PR button is initialized from data that comes from `wc_stripe_payment_request_params` and the other where the data comes from `get_cart_details`.

I think we should decide which behavior suits better for all cases and apply it to the product page and cart page (and other pages) to improve maintainability.

The are some considerations though. The first time the PR button is rendered for both the product page and the cart page (an other pages that could contain the PR button) would be totally unnecessary since the amount will be replaced later after the button is clicked. Te main reason to render the button when the page loads without making a request to get its details would be to allow users to click the button right away. 

I think we need to discuss it in another issue so we can mention all the pros and cons of consolidating this behavior.

Reply:
I don't think it's possible to make the button appear and be clickable without first initializing the payment request object.

Since we can't add the product to the cart on page load, I think we can't get the cart totals the same way in the product and cart pages.

If you have a different idea of an approach, let's create an issue then 👍